clustering/gke.py

```diff
@@ -0,0 +1,366 @@
+    body = {'cluster': cluster}
+    return body
+
+
+def main():
+    module = AnsibleModule(
+        argument_spec=dict(
+            credential_file=dict(required=True),
+            project_id=dict(required=True),
+
+            name=dict(required=True),
+            description=dict(required=False,
+                             default="GKE Cluster created with Ansible"),
+            node_count=dict(required=False, default=2),
+            node_config_machine_type=dict(required=False,
+                                          default="n1-standard-1"),
+            node_config_disksize=dict(required=False, default=100),
+            node_config_scopes=dict(required=False, default=[
+                "https://www.googleapis.com/auth/compute",
+                "https://www.googleapis.com/auth/devstorage.read_only",
+                "https://www.googleapis.com/auth/logging.write",
+                "https://www.googleapis.com/auth/monitoring"
+                ]
+            ),
+            node_config_metadata=dict(required=False),
+            username=dict(required=False, default="admin"),
+            password=dict(required=True),
+            logging_service=dict(required=False,
+                                 default="logging.googleapis.com"),
+            monitoring_svc=dict(required=False,
+                                default="monitoring.googleapis.com"),
+            network_name=dict(required=False, default="default"),
+            ipv4_range=dict(required=False),
+            wait_for=dict(required=False, default=True),
+            wait_poll_interval=dict(required=False, default=10),
+            zone=dict(required=True),
+            state=dict(default="present", choices=["present", "absent"])
+        )
+    )
+
+    if not HAS_GOOGLE_API_CLIENT:
+        module.fail_json(msg='This module requires google-api-python-client.')
+
+    creds_file = open(module.params.get("credential_file"), "r")
+    cred_data = json.load(creds_file)
+    credentials = SignedJwtAssertionCredentials(cred_data["client_email"],
+                                                cred_data["private_key"],
+                                                "https://www.googleapis.com/"
+                                                "auth/cloud-platform")
+
+    http = Http()
+    http = set_user_agent(http, USER_AGENT)
+    credentials.authorize(http)
+    gke = build("container", "v1", http=http)
+
+    changed = False
+    state = module.params.get("state")
+    project = module.params.get("project_id")
+    zone = module.params.get("zone")
+    name = module.params.get("name")
+    wait_for = module.params.get("wait_for")
+    wait_poll_interval = module.params.get("wait_poll_interval")
+    polling = wait_for
+
+    cluster_func = gke.projects().zones().clusters
+
+    if state == "present":
+        body = _make_cluster_body(module)
+        try:
+            resp = cluster_func().create(projectId=project, zone=zone,
```

Comment:
Is this idempotent? Are you just relying on the service call making sure it doesn't recreate anything necessary, and will this update the cluster to match the required config if it does already exist?